File: 1 to 100/35_search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        found = False
        for i in nums:
            if i == target:
                found = True
        if found:
            return nums.index(target)
        else:
            nums.append(target)
            nums.sort()
            return nums.index(target)
        

File: 1 to 100/35_search-insert-position.cpp
#include<bits/stdc++.h>
using namespace std;
#define endl '\n'
#define int long long
const int mod = 1e9+7;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int high = nums.size()-1;
        int low = 0;
        while(low<=high){
            int mid = (high+low)/2;
            if(nums[mid] == target){
                return mid;
            }else if(nums[mid] > target){
                high = mid - 1;
            }else low = mid + 1;
        }
        return low;
    }
};

signed main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);cout.tie(0);
    Solution obj;
}